src/cosmosShell/CosmosShellExtension.ts

```diff
@@ -0,0 +1,123 @@
+            vscode.commands.executeCommand(
+                'setContext',
+                'vscodeDatabases.cosmosShellSupportEnabled',
+                isCosmosShellInstalled,
+            );
+            registerCommandWithTreeNodeUnwrapping('cosmosDB.launchCosmosShell', launchCosmosShell);
+            if (isCosmosShellInstalled) {
+                ext.outputChannel.appendLine(`Cosmos Shell Extension: activated.`);
+            } else {
+                ext.outputChannel.appendLine(`Cosmos Shell Extension: deactivated.`);
+            }
+        });
+    }
+}
+
+function getCosmosShellCommand(): string {
+    const config = vscode.workspace.getConfiguration();
+    const shellPath: string | undefined = config.get('cosmosDB.shell.path');
+    return shellPath || 'CosmosShell';
+}
+
+export function launchCosmosShell(_context: IActionContext, node?: NoSqlContainerResourceItem) {
+    const isCosmosShellInstalled: boolean = isCosmosShellSupportEnabled();
+
+    if (!isCosmosShellInstalled) {
+        void vscode.window.showErrorMessage(
+            l10n.t(
+                'Cosmos Shell is not installed or not found in PATH. Please install Cosmos Shell or configure its path in settings.',
+            ),
+        );
+        return;
+    }
+
+    const command = getCosmosShellCommand();
+    const foundTerminal = vscode.window.terminals.find((terminal) => terminal.creationOptions.name === 'Cosmos Shell');
+
+    const config = vscode.workspace.getConfiguration();
+
+    const mcpEnabled = config.get<boolean>('cosmosDB.shell.mcp.enabled') ?? true;
+    const mcpPort = (config.get<number>('cosmosDB.shell.mcp.port') ?? 6128).toString();
+
+    const useMcp = mcpEnabled && !foundTerminal;
+    ext.outputChannel.appendLine(`MCP enabled: ${useMcp}, MCP port: ${mcpPort}`);
+    let args: string[];
+    if (!node) {
+        if (useMcp) {
+            args = ['--mcp', '--mcp-port', mcpPort];
+        } else {
+            args = [];
+        }
+        const terminal: vscode.Terminal = vscode.window.createTerminal('Cosmos Shell', command, args);
+        terminal.show();
+        return;
+    }
+    // connection string discovery for these items can be slow, so we need to run it with a temporary description
+    const rawConnectionString = node.model.accountInfo.endpoint;
+
+    if (!rawConnectionString) {
+        void vscode.window.showErrorMessage(l10n.t('Failed to extract the connection string from the selected node.'));
+        return;
+    }
+
+    if (useMcp) {
+        args = ['--mcp', '--mcp-port', mcpPort, '--connect', rawConnectionString];
+    } else {
+        args = ['--connect', rawConnectionString];
+    }
+    const terminal: vscode.Terminal = vscode.window.createTerminal('Cosmos Shell', command, args);
+
+    terminal.show();
+    if (node.model.container) {
+        terminal.sendText('cd ' + node.model.database.id + '/' + node.model.container.id, true);
```

Comment:
Shouldn't this pass a CR at the end? For me the cd command doesn't get executed but only entered:
<img width="358" height="159" alt="image" src="https://github.com/user-attachments/assets/bea40cfd-4008-4e3a-a42f-c67e9961c896" />

Reply:
On windows it works without.  Interesting.